python/lsst/ctrl/bps/clustered_quantum_graph.py

```diff
@@ -23,65 +23,477 @@
 a QuantumGraph.
 """
 
-__all__ = ["ClusteredQuantumGraph"]
+__all__ = ["QuantaCluster", "ClusteredQuantumGraph"]
 
 
-import networkx
+import logging
+import re
+import pickle
+from collections import Counter, defaultdict
+from networkx import DiGraph
+from pathlib import Path
 
+from lsst.daf.butler import DimensionUniverse
+from lsst.daf.butler.core.utils import iterable
+from lsst.pipe.base import QuantumGraph, NodeId
+from .bps_draw import draw_networkx_dot
 
-class ClusteredQuantumGraph(networkx.DiGraph):
+
+_LOG = logging.getLogger(__name__)
+
+
+class QuantaCluster:
+    """
+    name: `str`
+        Lookup key (logical file name) of file/directory. Must
+        be unique within ClusteredQuantumGraph.
+
```

Comment:
There should be no blank lines between parameter descriptions. (Please fix in all places where necessary. I won't be pointing these out individually.)